src/main/groovy/com/google/protobuf/gradle/ProtobufExtract.groovy

```diff
@@ -164,4 +156,48 @@ abstract class ProtobufExtract extends DefaultTask {
     }
     return new ArchiveActionFacade.ProjectBased(project)
   }
+
+  private FileCollection instantiateFilteredProtoDirs() {
+    return inputFiles.filter { File file -> file.isDirectory() }
```

Comment:
Why do we need to differentiate directory and non-directory cases? Can't we just include all the filtering logic in a single FileCollection?

Reply:
Copy spec needs dirs as inputs, while input snapshotting needs files from dirs. However, there is really no need to have dirs/non-dirs inputs, we can just apply filtering logic when inputs are computed, which I've done in ea8a1a3,